proxy/proxy_test.go

```diff
@@ -0,0 +1,65 @@
+// Copyright 2021 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package proxy
+
+import (
+	"errors"
+	"os"
+	"strings"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+)
+
+func checkValueFromEnvObj(name string, envVars []corev1.EnvVar) (string, error) {
+	for i := range envVars {
+		if envVars[i].Name == name {
+			return envVars[i].Value, nil
+		}
+	}
+	return "", errors.New("empty name")
+
+}
+
+var _ = Describe("Retrieving", func() {
+	Describe("proxy environment variables", func() {
+		// TODO(asmacdo) seems like a bad idea to mess with the env
```

Comment:
Hold placed to remove this after I get some feedback, the unit test is currently weak.